Mage/src/main/java/mage/game/stack/Spell.java

```diff
@@ -421,6 +422,7 @@ public boolean resolve(Game game) {
         } else {
             MageObjectReference mor = new MageObjectReference(getSpellAbility());
             game.storePermanentCostsTags(mor, getSpellAbility());
+            WarpAbility.addDelayedTrigger(getSpellAbility(), game);
```

Comment:
Smells bad. Just add additional effect with trigger create to warp’s spell ability in constructor. Example from awaken ability:

![IMG_1679](https://github.com/user-attachments/assets/42284786-8fc5-4866-aae6-7ddce56cf0d7)

Reply:
doesn't work on permanent spells, the spell ability doesn't execute its effects